python/pyarrow/tests/test_dataset.py

```diff
@@ -575,7 +575,7 @@ def test_expression_construction():
         field.isin(1)
 
     with pytest.raises(pa.ArrowInvalid):
-        field != {1}
+        field != object()
```

Comment:
Previously it was using `set` because we didn't support converting sets to scalar values. For `field != [1]` the code didn't raise because we supported converting lists to scalars.

Now we also support converting `sets` and thus we need to use a different entity to actually trigger an invalid filter.